internal/runner/command_test.go

```diff
@@ -0,0 +1,132 @@
+package runner
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/kballard/go-shellquote"
+)
+
+func TestRunAndForwardSignal(t *testing.T) {
+	cmd := exec.Command("echo", "hello world")
+
+	err := runAndForwardSignal(cmd)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestRunAndForwardSignal_CommandExitsWithNonZero(t *testing.T) {
+	cmd := exec.Command("false")
+
+	err := runAndForwardSignal(cmd)
+	exitError := new(exec.ExitError)
+	if errors.As(err, &exitError) && exitError.ExitCode() != 1 {
+		t.Errorf("Expected exit code 1, but got %d", exitError.ExitCode())
```

Comment:
Would it not also be a problem if `err` wasn't an `*exec.ExitError` (or unwrap to one)?

```suggestion
	if !errors.As(err, &exitError) {
		t.Fatalf("runAndForwardSignal(cmd) error type = %T (%v), want  *exec.ExitError", err, err)
	}
	if exitError.ExitCode() != 1 {
		t.Errorf("exitError.ExitCode() = %d, want 1", exitError.ExitCode())
```

(Or similar. Note I used `t.Fatalf` in order to avoid calling `exitError.ExitCode` if `errors.As` fails, because that would probably panic.)

Reply:
That's a good point!